airlock/api.py

```diff
@@ -316,30 +319,22 @@ def set_status(
 
         # validate first
         self.check_status(release_request, to_status, user)
-
-        # Ensure the state of this object is updated to reflect the change in status
-        # This is deliberately bypassing the frozen aspect of the class to keep
-        # things consistent. It does change the hash.
-        release_request.__dict__["status"] = to_status
+        release_request.status = to_status
 
     def add_file_to_request(
         self,
         release_request: ReleaseRequest,
         relpath: Path,
         user: User,
         group_name: Optional[str] = "default",
-    ) -> ReleaseRequest:
+    ):
         """Add a file to a request.
 
         Subclasses should do what they need to create the filegroup and
         record the file metadata as needed and THEN
```

Comment:
Should there be a comment that subclasses should also update the filegroups on this release request?

Reply:
Sorry, automerge stuck again.

There should, I will add in next PR